lib/muchos/azure.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+from sys import exit
+from .util import AMI_HELP_MSG, get_block_device_map
+from os.path import isfile
+import time
+from .existing import ExistingCluster
+import subprocess
+from os.path import isfile, join
+from azure.common.client_factory import get_client_from_auth_file
+from azure.mgmt.compute import ComputeManagementClient
+
+class VmssCluster(ExistingCluster):
```

Comment:
Is Vmss a Virtual Machine Scale Set?  

https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/overview

Reply:
That's correct. Virtual machine scale sets are the most efficient way to provision large virtual machine clusters in Azure.